Source/WebCore/page/Navigator.cpp

```diff
@@ -209,12 +209,16 @@ void Navigator::showShareData(ExceptionOr<ShareDataWithParsedURL&> readData, Ref
     if (!frame || !frame->page())
         return;
 
+    m_hasPendingShare = true;
+
     if (frame->page()->isControlledByAutomation()) {
-        promise->resolve();
+        RunLoop::main().dispatch([promise = WTFMove(promise), this] {
```

Comment:
There is no guarantee that `this` is still alive by the time the lambda runs (asynchronously). You should capture `weakThis = WeakPtr { *this }` instead.

Reply:
ah, good point.